### PR TITLE
[JENKINS-68639] Check whether an icon path already contains the context path before appending the context path to it

### PR DESCRIPTION
**Pull request by Riliane**
Check whether an icon path already contains the context path before appending the context path to it

See [JENKINS-68639](https://issues.jenkins.io/browse/JENKINS-68639).

Checking if the path already contains the context path for cases like implementations of `StatusIcon.getImageOf()` returning a path with the context path already, to avoid its duplication in the URL.

The `tryGetIconPath` method was not tested. The part buiding the resulting image path has been extracted for testability.

### Proposed changelog entries

* Fix raster image build status icons no longer displaying when running Jenkins with a non-empty context path due to a malformed URL (issue JENKINS-68639)

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change) and are in the imperative mood. [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [N/A] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadoc, as appropriate. 
- [N/A] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@Vlatombe @aHenryJard @janfaracik

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are accurate, human-readable, and in the imperative mood
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/6616"></a>

### CODE REVIEW

**Review by aneveux — APPROVED** [review]

**Review by Vlatombe — APPROVED** [review]

**Review by twasyl — APPROVED** [review]

**Review by bmunozm — APPROVED** [review]

**Comment by daniel-beck** [comment]
Would this still misbehave if the context path happens to be the start of the intended path?

**Comment by Vlatombe** [comment]
@daniel-beck Not sure what you mean exactly. I had in mind some edge cases (context path `/images` or `/plugin`) which I fixed and added coverage for. Do you have anything else in mind?

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
